FirebaseFunctions/Tests/SwiftIntegration/IntegrationTests.swift

```diff
@@ -0,0 +1,264 @@
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable("scalarTest")
+    XCTAssertNotNil(function)
+    function.call(17 as Int16) { result, error in
+      do {
+        XCTAssertNil(error)
+        let data = try XCTUnwrap(result?.data as? Int)
+        XCTAssertEqual(data, 76)
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result: \(error)")
+      }
+    }
+    waitForExpectations(timeout: 1)
+  }
+
+  func testToken() {
+    // Recreate _functions with a token.
+    let functions = FunctionsFake(
+      projectID: "functions-integration-test",
+      region: "us-central1",
+      customDomain: nil,
+      withToken: "token"
+    )
+    functions.useLocalhost()
+
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable("FCMTokenTest")
+    XCTAssertNotNil(function)
+    function.call([:]) { result, error in
+      do {
+        XCTAssertNil(error)
+        let data = try XCTUnwrap(result?.data) as? [String: Int]
+        XCTAssertEqual(data, [:])
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result: \(error)")
+      }
+    }
+    waitForExpectations(timeout: 1)
+  }
+
+  func testFCMToken() {
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable("FCMTokenTest")
+    XCTAssertNotNil(function)
+    function.call([:]) { result, error in
+      do {
+        XCTAssertNil(error)
+        let data = try XCTUnwrap(result?.data) as? [String: Int]
+        XCTAssertEqual(data, [:])
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result: \(error)")
+      }
+    }
+    waitForExpectations(timeout: 1)
+  }
+
+  func testNull() {
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable("nullTest")
+    XCTAssertNotNil(function)
+    function.call(nil) { result, error in
+      do {
+        XCTAssertNil(error)
+        let data = try XCTUnwrap(result?.data) as? NSNull
+        XCTAssertEqual(data, NSNull())
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result: \(error)")
+      }
+    }
+    waitForExpectations(timeout: 1)
+  }
+
+  func testMissingResult() {
+    let expectation = expectation(description: #function)
+    let function = functions.httpsCallable("missingResultTest")
+    XCTAssertNotNil(function)
+    function.call(nil) { result, error in
+      do {
+        XCTAssertNotNil(error)
+        let error = try XCTUnwrap(error) as NSError
+        XCTAssertEqual(FunctionsErrorCode.internal.rawValue, error.code)
+        XCTAssertEqual("Response is missing data field.", error.localizedDescription)
+        expectation.fulfill()
+      } catch {
+        XCTAssert(false, "Failed to unwrap the function result: \(error)")
+      }
```

Comment:
LGTM. You can avoid the `do {} catch {}` blocks by adding `throws` to the test method signature. If the `XCTUnwrap` fails, it'll highlight the line with the failing unwrap.

Reply:
I don't think it will work if `XCTUnwrap` is used inside of the closure.

Reply:
Right @maksymmalyhin That's what I found:

![Screen Shot 2021-11-11 at 8 40 56 AM](https://user-images.githubusercontent.com/73870/141335571-a12b3c79-b4e0-45fe-affd-4de1f89b4b21.png)

Reply:
Oh, never thought about that. Makes sense, thanks for pointing that out y'all!